framework/src/modules/random/stores/used_hash_onions.ts

```diff
@@ -100,6 +100,16 @@ export class UsedHashOnionsStore extends BaseOffchainStore<UsedHashOnionStoreObj
 
 		if (index === -1) {
 			originalUsedHashOnions.push(usedHashOnion);
+			// if newly inserted entry has count 0, then cleanup previous entries
+			if (usedHashOnion.count === 0) {
+				const newUsedHashonions = [usedHashOnion];
+				await this.set(
+					ctx,
+					address,
+					this._filterUsedHashOnions(newUsedHashonions, finalizedHeight),
+				);
+				return;
+			}
```

Comment:
Here you can slightly improve the code by moving your if statement before `originalUsedHashOnions.push(usedHashOnion)`. If count is 0, then there is no need to modify `originalUsedHashOnions` array.

But even better, what if instead of duplicating the code with the second `this.set()` followed by a `return`, your conditional would simply set value for a variable containing used hash onions? Then the one and only `this.set()` at the end of the method would simply use that variable, which would be common for all the possible paths?

I personally think that's a cleaner approach.

Reply:
1. moving the if statement before `originalUsedHashOnions.push(usedHashOnion)` made the logic more efficient, yes. Thank you.

2. If we apply the second part of your suggestion the code would look like this :
```
		let newUsedHashOnions: UsedHashOnion[];
		if (index === -1) {
			// if newly inserted entry has count 0, then cleanup previous entries
			if (usedHashOnion.count === 0) {
				newUsedHashOnions = [usedHashOnion];
				await this.set(
					ctx,
					address,
					this._filterUsedHashOnions(newUsedHashOnions, finalizedHeight),
				);
			}
			else {
				originalUsedHashOnions.push(usedHashOnion);
				newUsedHashOnions = originalUsedHashOnions;
			}
		} else {
			// eslint-disable-next-line no-param-reassign
			originalUsedHashOnions[index] = usedHashOnion;
			newUsedHashOnions = originalUsedHashOnions;
		}

		await this.set(
			ctx,
			address,
			this._filterUsedHashOnions(newUsedHashOnions, finalizedHeight),
		);
```

In terms of computation we are getting rid of 1 `return` call at the cost of 1 conditional `else` branch. Not specifically sure about nodejs but with most compilers this modification would be less efficient. However if you know that it's not true I'd then update the PR per your suggestion.